src/components/Header.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { useDispatchContext, useStateContext } from './GlobalStateProvider';
+
+import * as todoService from '../api/todos';
+import { Todo } from '../types/Todo';
+
+export const Header: React.FC = () => {
+  const { title, todos } = useStateContext();
+  const dispatch = useDispatchContext();
+
+  const [adding, setAdding] = React.useState(false);
```

Comment:
Rename all boolean variables using such [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

```suggestion
  const [isAdding, setIsAdding] = React.useState(false);
```